### ISSUE TITLE
arg test fails

### ISSUE THREAD

**Issue by snacsnoc** [dev-sushi/sushi#15]
Arg test fails due to a syntax error in the function call using a positional argument after a keyword argument.
I'm going to assume 223a859 commit acknowledges this error, but please correct me if I'm wrong.

Replicate:
`python3.9 examples/arg-cpp/app.py`

Error:
```
sushi   saving indexed functions (lib/main.hpp)
Traceback (most recent call last):
  File "/Users/easto/Library/Mobile Documents/com~apple~CloudDocs/Projects/sushi/examples/arg-cpp/app.py", line 10, in <module>
    from out.main import helloWorld
  File "/Users/easto/Library/Mobile Documents/com~apple~CloudDocs/Projects/sushi/examples/arg-cpp/out/main.py", line 2
    def helloWorld(number):     Execute(file='main.hpp', number)
                                                           ^
SyntaxError: positional argument follows keyword argument

```

Code in question: https://github.com/dev-sushi/sushi/blob/main/src/sushipy/index.py#L118

With an argument, this will generate an error in the code:

```
from sushipy.execute import Execute
def helloWorld():	Execute(file='main.hpp', number)
```

Solution:
`index.py` should write keyword arguments by default. Positional arguments might be easier to parse.

**Comment by ksawery29** [comment]
It looks like it. I will fix that issue later today. It's strange that it was working on my pc

**Comment by ksawery29** [comment]
@snacsnoc I merged your PR but the issue is still here. Still confused because its working on my pc

**Comment by snacsnoc** [comment]
Python 3.9. 

The error was present with sushi version 0.1.1, I see that you have now updated the package to 0.1.2. With PR #16, this specific issue with keyword arguments has been fixed.

I'm still not able to run the arg-cpp example. Because `examples/arg-cpp/lib/main.hpp` does not have a main() function, g++ throws an error:
```
(base) easto@Eastons-MacBook-Air arg-cpp % python app.py
sushi   saving indexed functions (lib/main.hpp)
Undefined symbols for architecture arm64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture arm64
collect2: error: ld returned 1 exit status
Traceback (most recent call last):
  File "/Users/easto/Library/Mobile Documents/com~apple~CloudDocs/Projects/sushi/examples/arg-cpp/app.py", line 12, in <module>
    helloWorld(1)
  File "/Users/easto/Library/Mobile Documents/com~apple~CloudDocs/Projects/sushi/examples/arg-cpp/out/main.py", line 2, in helloWorld
    def helloWorld(number):     Execute('main.hpp', number)
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/site-packages/sushipy/execute.py", line 80, in __init__
    self.function()
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/site-packages/sushipy/execute.py", line 117, in function
    subprocess.call([f"./{path}/out"], shell=False)
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/subprocess.py", line 349, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
OSError: [Errno 8] Exec format error: './lib/out'
```
Adding the `-c` switch to g++ allows it compile by not running the linker.
modified sushi.conf:
```
[launch]
exec_command = g++ -c -o lib/out [file-name]
```

Once I do that, I'm still not able to run the arg-cpp example. Output:
```
(base) easto@Eastons-MacBook-Air arg-cpp % python app.py
sushi   saving indexed functions (lib/main.hpp)
Traceback (most recent call last):
  File "/Users/easto/Library/Mobile Documents/com~apple~CloudDocs/Projects/sushi/examples/arg-cpp/app.py", line 12, in <module>
    helloWorld(1)
  File "/Users/easto/Library/Mobile Documents/com~apple~CloudDocs/Projects/sushi/examples/arg-cpp/out/main.py", line 2, in helloWorld
    def helloWorld(number):     Execute('main.hpp', number)
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/site-packages/sushipy/execute.py", line 80, in __init__
    self.function()
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/site-packages/sushipy/execute.py", line 117, in function
    subprocess.call([f"./{path}/out"], shell=False)
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/subprocess.py", line 349, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
OSError: [Errno 8] Exec format error: './lib/out'

```
The compiled file:
```
(base) easto@Eastons-MacBook-Air arg-cpp % file lib/out 
lib/out: Mach-O 64-bit object arm64
```

This may be more related to my build system (arm64), I will try on an amd64 system.

**Comment by ksawery29** [comment]
I looked more into the error that you sent and it might be arm64 related issue. Main isn't required for .hpp files, only for .cpp files (which is used [here](https://github.com/dev-sushi/sushi/blob/0cb30adbb6c9ee7979ed78ea9e421d5e9f699ada/examples/arg-cpp/sushi.conf#L10)) but I'm not sure. 

The compiling works for me (x64) though.

**Comment by ksawery29** [comment]
@snacsnoc I think I found the issue. After some testing with [nektos/act](https://github.com/nektos/act) I found out that nothing was writing to the temporary file. See this:
```
# cat lib/main.hpp
void helloWorld(int number)
{
    std::cout << "You provided " << number << "\n";
}#     
# 
# cat lib/temp.cpp
#
```

That's why it was throwing error about no main function.

**Comment by ksawery29** [comment]
Fixed! It turns out that instead of `TEMP_FILE` there should be `self.temp_file` ([here](https://github.com/dev-sushi/sushi/blob/19a72902f3931911040a5a3b4c54e3f7f627a04d/src/sushipy/execute.py#L97l)). But we will need to revert #16 pull request because sushi takes the file name as an argument also. 
```
lib/temp.cpp:2:35: error: expected ‘)’ before numeric constant
    2 |  int main() {helloWorld('main.hpp' 1);}
```

After reverting that it works without any issue, but if you think we should keep it then let me know.

**Comment by snacsnoc** [comment]
Thanks for taking the time to investigate, I'll will attempt to fix this and submit a PR. The goal of PR #16 was to fix the arguments in the generated Python 'out' file.

I will re-write [this](https://github.com/dev-sushi/sushi/blob/19a72902f3931911040a5a3b4c54e3f7f627a04d/src/sushipy/index.py#L116), thanks for looking at it!

**Comment by snacsnoc** [comment]
It is also missing the comma between the filename and argument:
```
lib/temp.cpp:2:25: warning: character constant too long for its type
    2 |  int main() {helloWorld('main.hpp' 1);}
      |                         ^~~~~~~~~~
lib/temp.cpp: In function 'int main()':
lib/temp.cpp:2:35: error: expected ')' before numeric constant
    2 |  int main() {helloWorld('main.hpp' 1);}
      |                        ~          ^~
      |                                   )
```

**Comment by ksawery29** [comment]
Yeah, that's what i'm saying. But the file name shouldn't be passed down to cpp. File name is just a note for sushi in what file was it executed.

**Comment by snacsnoc** [comment]
@ksawery29 Got it working! I'll submit a PR soon

**Comment by ksawery29** [comment]
> 
> 
> I'll track down that missing semicolon (did you mean comma?)

@snacsnoc yes, I meant comma. Sorry for that mistake haha

**Comment by snacsnoc** [comment]
@ksawery29 No problem at all. This is fixed in PR #17

**Comment by snacsnoc** [comment]
@ksawery29 This has been fixed, but as per the discussion on PR #17, your suggestion it to use positional then keyword arguments (file=) when writing to our output file. Can we close this issue and open a new one?

**Comment by ksawery29** [comment]
Sure! Just make sure that title isn't the same.
